doc: add documentation framework and basic docs (#172)
NordicSemiconductor/pc-nrfconnect-npm: doc/docs/overview.md

       ![Example of a tooltip in nPM PowerUP](./screenshots/npm_tooltip_example.png "Example of a tooltip in nPM PowerUP")

## Before selection

Before a device is selected, the side panel contains the following buttons:

### Select Device

Dropdown to list the PMIC devices attached to the computer.

!!! note "Note"
      Read [Connect the nPM1300 EK with nPM PowerUP](https://docs.nordicsemi.com/bundle/ug_npm1300_ek/page/UG/nPM1300_EK/use_ek_power_up.html) for information about the hardware setup required to use the nPM1300 EK with nPM PowerUP.

## After selection

When you select a device, the application checks the firmware programmed on the nPM Controller. If needed, it asks you to program the required firmware.
Then, the side panel options become available.

![nPM PowerUP application window with a device selected](./screenshots/npm_overview_device_selected.png "nPM PowerUP application window with a device selected")

### Actions

This side panel area contains the following buttons:

|          Button          | Description |
| ------------------------ | ----------- |

> | **Export Configuration** | Export the PMIC configuration based on the nPM PowerUP application settings. You can save the configuration to an `.overlay` file for [use in the nRF Connect SDK](https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/device_guides/pmic/npm1300.html#importing_an_overlay_from_npm_powerup) or to a JSON file for later use in nPM PowerUP.</br></br>You can also set the configuration before you select a device and export it to a file. |

Reviewer: ```suggestion
| **Export Configuration** | Export the PMIC configuration based on the nPM PowerUP application settings. You can save the configuration to an `.overlay` file for [use in the nRF Connect SDK](https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/device_guides/pmic/npm1300.html#importing_an_overlay_from_npm_powerup) or to a JSON file for later use in nPM PowerUP.</br></br>You can also set the configuration before you select a device and export it to a file ( offline mode ). |
```

> | **Load Configuration**   | Load the PMIC configuration from a JSON file and update all configurations accordingly.</br></br>You can also load a configuration before you select a device.  |

Reviewer: ```suggestion
| **Load Configuration**   | Load the PMIC configuration from a JSON file and update all configurations accordingly.</br></br>You can also load a configuration without selecting a device ( offline mode ).  |
```

| **Reset Device**         | Reset the PMIC device and the nPM Controller. The PMIC default device configuration is restored.  |
| **Record Events**        | Record all terminal [log](#log) events to CSV files in a selected directory. |

### Fuel Gauge

This side panel area lets you select the following options:

|               Menu               | Description |
| -------------------------------- | ----------- |
| **Active Battery Model**         | Select the battery model you want to use for Fuel Gauge in nPM PowerUP.        |
| **Add New Active Battery Model** | Select a battery from selected vendors that has been profiled by Nordic Semiconductor or your own custom battery model, added with the **Profile Battery** feature and saved in the [**Profiles** tab](#profiles-tab).        |
| **Profile Battery**              | Create your own, custom battery profile, and collect the data. These are then saved in the [**Profiles** tab](#profiles-tab) and added to the **Add New Active Battery Model** drop-down menu.</br></br>An additional board, nPM Fuel Gauge, is required to perform the battery profiling.        |

### Settings

> This side panel area lets you configure the reporting rate.

Reviewer: ```suggestion
This side panel area lets you configure the reporting rate. This effect the frequency of the data on the graph pane and the battery status card
```

### Connection Status

You can check the connection status to the [device you selected](#select-device).

![Example of Connection Status after selecting a device](./screenshots/npm_connection_status.PNG "Example of Connection Status after selecting a device")

## Dashboard tab

The **Dashboard** tab provides a quick look overview of the major PMIC settings from the **Charger** and **Regulators** tabs.

!!! info "Tip"
      Use the expand button in the top-right corner of some of the tiles to see all the options of the given setting.

## Charger tab

You can use the options in the **Charger** tab to control and monitor the charging settings and status of the PMIC device.

!!! info "Tip"
      Check the battery datasheet for the termination voltage and current limit values that you can set for your battery.

> Using the built-in battery models, you can get an estimated time-to-full and time-to-empty when charging or discharging a battery connected to the EK.

Reviewer: ```suggestion
Using the built-in battery models, you can get an estimated time-to-full and time-to-empty when charging or discharging a battery connected to the EK. This also provides a State of Chard in percentage (SOC) 
```

| **Create New Project**   | Create a new custom project template. This option includes some of the fields present when using the [**Profile Battery** option](#fuel-gauge), but without collecting data. |

The following drop-down menu options are available for the project and profile sections:

|          Button          | Area     |                                                             Description                                                       |
| ------------------------ | -------- | ----------------------------------------------------------------------------------------------------------------------------- |
| **Process Included Data**| Project  | If your project includes multiple temperature profiles, you can include a selection of them with the **Include** toggle. Using this option processes only the included temperature profiles. |
| **Save Battery Model**   | Project  | Save the battery model for the entire project to a JSON or an INC file. This only saves the temperature profiles you have selected with the **Include** toggle. |

> | **Write Battery Model**  | Project  | Write the battery model for the entire project to the nPM Controller. This only writes the temperature profiles you have selected with the **Include** toggle. |

Reviewer: ```suggestion
| **Write Battery Model**  | Project  | Write the battery model to the EK for the entire project to the nPM Controller. This only writes the temperature profiles you have selected with the **Include** toggle. |
```

| **Add Profile**          | Project  | Add a new battery profile with a specific temperature and a CSV data file. Data files are generated when you use the [**Profile Battery** option](#fuel-gauge). |
| **Open Folder**          | Project  | Open the project directory in the **File Explorer**. |
| **Edit Project**         | Project  | Edit the project settings.  |
| **Remove Project**       | Project  | Remove the current project, including all battery profiles. |
| **Include** toggle       | Profile  | Select the temperatures to be included for processing with **Process Included Data** button for the entire project. |
| **Process Data**         | Profile  | Process the data for the given temperature profile only. |
| **Save Battery Model**   | Profile  | Save the battery model to a JSON or an INC file. |
| **Write Battery Model**  | Profile  | Write the battery model to the nPM Controller. |
| **Edit Profile**         | Profile  | Edit the battery profile. |
| **Remove Profile**       | Profile  | Remove the battery profile.|

### Bundled profiles

Here you write and save battery models from selected vendors that has been profiled by Nordic Semiconductor.

> When writing, you are going to write new firmware on the nPM Controller to one of its available battery model slots.

Reviewer: ```suggestion
When writing, you are going to write new battery model on the nPM Controller to one of its available battery model slots.
```

This can overwrite the [**Active Battery Model**](#fuel-gauge).

> When saving, the battery model is saved in the INC file format in the selected directory. This file format is meant for integrating the battery model into your final application with a Nordic System on Chip (SoC).

Reviewer: ```suggestion
When saving, the battery model is saved in the JSON that can be downloaded to the nPM Controller or INC file format in the selected directory. This file format is meant for integrating the battery model into your final application with a Nordic System on Chip (SoC).
```

## Graph tab

> After generating the battery model, you can use the **Graph** tab to evaluate the battery state-of-charge predictions in real time. Make sure the battery Fuel Gauge in the **Dashboard** tab or **Fuel Gauge** tab is enabled.

Reviewer: ```suggestion
Here you can monitors the state of the PMIC over the such as current voltage temperature and State of Charge (SOC). For SOC make sure the battery Fuel Gauge in the **Dashboard** tab or **Fuel Gauge** tab is enabled.
```
